### PR TITLE
Update num of files during manifest update.

### PR DESCRIPTION
**Pull request by myersCody**
Smoke tests for GCP are failing due to the processing count being off in the sources stats endpoint.  

## Steps to reproduce on main: 
I was able to replicate the problem locally by using a combination of a dev gcp account, and date override. 

1) set the `DATE_OVERRIDE` variable to be a couple days ago. 
```
DATE_OVERRIDE='2021-09-07'
```
After you set the date override in the env, you will need to restart the worker and check the override date in the container:
```
(koku) bash-3.2$ docker-compose up -d koku-worker
WARNING: The RUN_GUNICORN variable is not set. Defaulting to a blank string.
WARNING: The USE_RABBIT variable is not set. Defaulting to a blank string.
WARNING: The SOURCES_API_PREFIX variable is not set. Defaulting to a blank string.
Docker Compose is now in the Docker CLI, try `docker compose up`

Starting koku_koku-base_1 ...
Starting koku_koku-base_1 ... done
koku_koku-worker_1 is up-to-date
(koku) bash-3.2$ docker exec -it koku_koku-worker_1 bash -c "echo $DATE_OVERRIDE"
2021-09-07
```
2) Then you will need to add a dev gcp account
```
make gcp-source gcp_name=cody-test
```
This will trigger the initial download.

3) Bump the date override to the next day: 
```
DATE_OVERRIDE='2021-09-08'
```
Note: you will need to reset the koku-worker and check the date with commands in step one. 
4) Hit the download endpoint
http://127.0.0.1:5000/api/cost-management/v1/download/
5) Check your source stats:
http://localhost:8000/api/cost-management/v1/sources/c82a8ebd-43f8-4b75-b7be-967392a5bc7b/stats/
```
 "files_processed": "3/2",
 ```
 
 ## How to test:
 Run the reproduction steps above for this branch and noticed that the files_processed has changed to:
```
 "files_processed": "3/3",
 ```

### CODE REVIEW

File: koku/masu/database/report_manifest_db_accessor.py [hunk 1]
```diff
@@ -51,6 +51,14 @@ def mark_manifest_as_completed(self, manifest):
             manifest.manifest_completed_datetime = self.date_accessor.today_with_timezone("UTC")
             manifest.save()
 
+    def update_number_of_files_for_manifest(self, manifest):
+        """Update the number of files for manifest."""
+        records = CostUsageReportStatus.objects.filter(manifest_id=manifest.id)
+        set_num_of_files = len(records)
```
Comment (maskarb) [inline koku/masu/database/report_manifest_db_accessor.py:57]: Can we use `.count()` here instead?

```
Note: If you only need to determine the number of records in the set (and don’t need 
the actual objects), it’s much more efficient to handle a count at the database level using 
SQL’s SELECT COUNT(*). Django provides a count() method for precisely this reason.
```
From [docs](https://docs.djangoproject.com/en/3.2/ref/models/querysets/).

Comment (myersCody) [reply koku/masu/database/report_manifest_db_accessor.py]: Done 👍

**Comment by myersCody** [comment]
Smoke Tests:
```
lib64.python3.6.site-packages.iqe_cost_management.tests.rest_api.v1.test__cost_model.test_api_cost_model_negative_unbound_tier[tiered_rate_unbound_start]
lib64.python3.6.site-packages.iqe_cost_management.tests.rest_api.v1.test__cost_model.test_api_cost_model_negative_unbound_tier[tiered_rate_unbound_end]
lib64.python3.6.site-packages.iqe_cost_management.tests.rest_api.v1.test_gcp_storage_reports.test_api_gcp_storage_report_filter_match_date_range_group_by[account-region-service-last-30-days]
```
After chatting with Luke these test failures seem unrelated to my changes.

**Review by dccurtis — APPROVED** [review]

**Comment by myersCody** [comment]
retest

**Pull request merged**